pkg/controller/clusterinstallation/minio.go

```diff
@@ -2,18 +2,105 @@ package clusterinstallation
 
 import (
 	"context"
+	"fmt"
+	"reflect"
 
 	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
 	mattermostv1alpha1 "github.com/mattermost/mattermost-operator/pkg/apis/mattermost/v1alpha1"
+	mattermostMinio "github.com/mattermost/mattermost-operator/pkg/components/minio"
+
+	minioOperator "github.com/minio/minio-operator/pkg/apis/miniocontroller/v1beta1"
 )
 
-func (r *ReconcileClusterInstallation) checkMinioDeployment(mattermost *mattermostv1alpha1.ClusterInstallation, reqLogger logr.Logger) error {
-	minioExist := false
+func (r *ReconcileClusterInstallation) createMinioDeployment(mattermost *mattermostv1alpha1.ClusterInstallation, deployment *minioOperator.MinioInstance, reqLogger logr.Logger) error {
```

Comment:
You should be able to use `func (r *ReconcileClusterInstallation) createResource` instead of this function